Metadata/CIP-Certification-Metadata.md

```diff
@@ -0,0 +1,180 @@
+
+
+
+## Motivation: why is this CIP necessary?
+<!-- A clear explanation that introduces the reason for a proposal, its use cases and stakeholders. If the CIP changes an established design then it must outline design issues that motivate a rework. For complex proposals, authors must write a Cardano Problem Statement (CPS) as defined in CIP-9999 and link to it as the `Motivation`. -->
+
+It is expected that evidence of various kinds of assurance of DApps is recorded in an immutable and verifiable way on the Cardano blockchain; these forms of assurance are: automated testing, including property-based testing (level 1), audit (level 2) and formal verification (level 3).
+
+The metadata should be discoverable by all certification stakeholders, including end-users, DApp developers, and ecosystem components, such as light wallets and DApp stores. Information should be indexable by certification provider, DApp developer, DApp and DApp version.
+
+### Use-cases and Stakeholders
+DApp developers will seek certification from one of the providers, according to their desired level of certification and will refer to the certificate on several platforms (e.g. on their website or in their DApp registration on a DApp store).
+
+Certification is to be provided by certification providers including: testing services (level 1), auditors (level 2), and verification services (level 3). Certification providers will issue these certificates refering to a particular version of a DApp that has succesfully gone through a certification process. 
+
+DApp stores and light wallets will be able to pull DApps information from DApps registration or chain exploring and will link a DApp to a corresponding set of certificates.
+
+End-user will interact with a DApp through a wallet and will be able to check the different certificates obtained by the DApp.
+
+
+## Specification
+<!-- The technical specification should describe the proposed improvement in sufficient technical detail. In particular, it should provide enough information that an implementation can be performed solely on the basis of the design in the CIP. This is necessary to facilitate multiple, interoperable implementations. -->
+
+### Certification providers
+Certification providers will present which level of certification was reached and present evidence of the work done. Certification providers will sign the certificate to attest that they have done the work and prevent certificate forgeries.
+
+
+### Definitions
+- **dApp** - A decentralised application that is described by the combination of metadata, certificate and a set of used scripts.
+- **dApp Store** - A dApp aggregator application which follows on-chain data looking for and verifying dApp metadata claims, serving their users linked dApp metadata.
+
+
+### Properties
```

Comment:
What about putting these fields as an array with:
- field name
- type
- mandatory?
- depiction

Reply:
Seeing that this comment led to confusion, let me rephrase. What about describing these properties in a Markdown table, with four columns, one for field name, one for type, one for whether it is mandatory and one for a description of the entry. For instance:

```markdown
| Field name                 | Type                        | Mandatory? | Description
|----------------------------|-----------------------------|------------|---------------
| `subject`                  | UTF-8 encoded string        | Yes        | Identifier of the claim subject (i.e dApp)
| `type`                     | JSON object                 | Yes        | The type of the claim
| `type.action`              | `CERTIFY` or `AUDIT`        | Yes        | The action that the certificate is asserting. In the case of `CERTIFY`, the certificate is asserting that the dApp is being certified. In the case of `AUDIT`, the certificate is asserting that an audit has been performed for this DApp. `AUDIT` shall come with a certification level `certificationLevel` of `0`. [IDEA: Revoke? Update?]
| `type.certificationLevel`  | Integer between `0` and `3` | Yes        | The level of certification. `0` is reserved for audits and reports that are not compliant with the certification standards. `1`, `2`, and `3` refer to the certification certificates referring to the three level of certifications as defined by the certification working group.
| `type.certificationIssuer` | UTF-8 encoded string        | Yes        | The name of the certification issuer.
```

I don't know how much better than would be. Alternatively, maybe a list formatted a bit more strictly:

```markdown
- `subject`: Identifier of the claim subject (i.e. dApp).
  - **mandatory**: yes
  - **type**: UTF-8 encoded string

- `type`: The type of the claim.
  - **mandatory**: yes
  - **type**: JSON object

- `type.action`: The action that the certificate is asserting.
  - **mandatory**: yes
  - **type**: `"CERTIFY"` or `"AUDIT"`
  - In the case of `CERTIFY`, the certificate is asserting that the dApp is being certified. In the case of `AUDIT`, the certificate is asserting that an audit has been performed for this DApp. `AUDIT` shall come with a certification level `certificationLevel` of `0`. [IDEA: Revoke? Update?]

- `type.certificationLevel`: Certification level.
  - **mandatory**: yes
  - **type**: integer between 0 and 3
  - 0 is reserved for audits and reports that are not compliant with the certification standards. 1, 2, and 3 refers to the certification certificates referring to the three level of certifications as defined by the certification working group.

- `certificationIssuer`: The name of the certification issuer.
  - **mandatory**: yes
  - **type**: UTF-8 encoded string
```

A pretty visualisation could otherwise simply be generated from the JSON schema.